pkg/pgclient/client.go

```diff
@@ -161,7 +166,7 @@ func (cfg *Config) GetNumConnections() (min int, max int, numCopiers int, err er
 
 	numCopiers = perProc * maxProcs
 	// we leave one connection per-core for non-copier usages
-	if numCopiers+maxProcs > max {
+	if numCopiers > max-maxProcs {
```

Comment:
if we're doing this, do you want to set the default `db-writer-connection-concurrency` back to `5`?